src/main/java/gov/cabinetoffice/gap/service/ZipService.java

```diff
@@ -109,19 +99,23 @@ private static void downloadFile(final String fileName) {
         }
     }
 
-    public static String parseFileName(final String fileName, int suffix) {
-        String[] subString = fileName.split("/");
-        logger.info("subString " + subString);
-        String[] parts = subString[subString.length - 1].split("\\.");
-        logger.info("parts " + parts);
-
-        String name = parts[0];
-        logger.info("name " + name);
-        String extension = parts.length > 1 ? "." + parts[1] : "";
-        logger.info("extension " + extension);
-
-        logger.info("name concat " + name.concat("_" + suffix + extension));
-        return name.concat("_" + suffix + extension);
+    public static String parseFileName(final String objectKey, int suffix, final String applicationId,
+                                       final String submissionId) {
+        //an object key is formed by applicationId/submissionId/s3bucketRandomFolderName/filename
+        final String applicationIdAndSubmissionId = applicationId + "/" + submissionId + "/";
+        final String filenameWithoutApplicationIdAndSubmissionId = objectKey.replace(applicationIdAndSubmissionId, "");
+        final String folderNameToRemove = filenameWithoutApplicationIdAndSubmissionId.split("/")[0];
+        final String filenameWithoutFolderName = filenameWithoutApplicationIdAndSubmissionId.replace(folderNameToRemove + "/", "");
+        //the file that we allow to attach can only be of the following types: .doc, .docx, .odt, .pdf, .xls, .xlsx, .zip
+        String filenameWithoutExtension = filenameWithoutFolderName.split("\\.(doc|docx|odt|pdf|xls|xlsx|zip)$")[0];
```

Comment:
Can we make this regex a constant? I have some concerns that if we change validation rules in the service we will also need to change the regex as well. Is there any way to do this without specifying file types?